packages/at_client/lib/src/service/sync_service_impl.dart

```diff
@@ -114,7 +114,9 @@ class SyncServiceImpl implements SyncService, AtSignChangeListener {
   }
 
   @override
-  void sync({Function? onDone, Function? onError}) {
+  void sync(
+      {@Deprecated('Use SyncProgressListener') Function? onDone,
```

Comment:
Please make sure the deprecated annotations are in both sync_service.dart and sync_service_impl.dart

Reply:
Added deprecated tag to onDone callback in sync_service.dart as well.